### PR TITLE
Perform Evaluation on RDD

### PR DESCRIPTION
**Pull request by saurfang**
Revert code clean up that merges the evaluation of records from RDD and List. Instead of merging the code path using List, perform evaluation on RDD instead to handle large dataset.

There is no reliable way to get a SparkContext especially with the way we set up our tests. Since there is really no need to run evaluation on List of records and lists can always be parallelize with single function call, we are removing evaluation on list altogether.

Related to #237 #231 

@jq @deerzq @luanjunyi

### CODE REVIEW

File: demo/twenty_news/build.gradle [hunk 1]
```diff
@@ -17,15 +17,10 @@ dependencies {
   compile 'org.slf4j:slf4j-simple:1.7.7'
 }
 
-compileScala {
```
Comment (luanjunyi) [inline demo/twenty_news/build.gradle]: Why remove these `compileScala`?

File: training/src/main/scala/com/airbnb/aerosolve/training/Evaluation.scala [hunk 2]
```diff
@@ -92,7 +83,16 @@ object Evaluation {
       .toArray
   }
 
-  def evaluateMulticlassClassification(records : RDD[EvaluationRecord]) : Array[(String, Double)] = {
+  def evaluateBinaryClassificationWithResults(records: List[EvaluationRecord],
+                                              buckets: Int,
+                                              evalMetric: String,
+                                              resultsOutputPath: String): Array[(String, Double)] = {
+    val rdd = SparkContext.getOrCreate().parallelize(records)
+    val result = evaluateBinaryClassificationWithResults(rdd, buckets, evalMetric, resultsOutputPath)
```
Comment (luanjunyi) [inline training/src/main/scala/com/airbnb/aerosolve/training/Evaluation.scala]: Let's simplify, just one line:

`evaluateBinaryClassificationWithResults(rdd, buckets, evalMetric, resultsOutputPath)`

File: demo/twenty_news/build.gradle (hunk 1 above)
Comment (saurfang) [inline demo/twenty_news/build.gradle]: The `fork` option has been removed in Gradle 3+ and is causing error in my local environment. https://docs.gradle.org/current/release-notes#ant-based-scala-compiler-has-been-removed

Comment (timyitong) [inline demo/twenty_news/build.gradle]: Side discussion, can we switch to gradle wrapper? This could ensure that everyone's using exactly same gradle to compile.

Comment (saurfang) [inline demo/twenty_news/build.gradle]: @timyitong  +1 if you send a PR ;)

**Comment by timyitong** [comment]
Why local spark context is "not reliable"? Is it due to thread safety issue? One thing I do know is that in local tests, all spark context related methods should be synchronized, local spark context is not thread safe; they may stomp on each other due to parallel test execution.

**Comment by saurfang** [comment]
@timyitong The issue I faced was `SparkContext.getOrCreate` uses a default Spark conf if it is creating a new one. The conf would initialize with system properties and in this case we have to set master and app name explicitly. 
The particular test case on List evaluation wasn't using any SparkContext at all and I don't feel this function should take a SparkContext, which would defeat the design (sort of). Then there was the good old issue where you can't have multiple spark sessions. 
I suppose the correct way to do it is to initialize a SparkContext explicitly in the test case even though it is evaluating on a `List`, then `SparkContext.getOrCreate` will just pick up that. 

Anyway, long story short, I don't feel there is particular value to have the List version so simplifying is probably good enough.

File: demo/twenty_news/build.gradle (hunk 1 above)
Comment (timyitong) [inline demo/twenty_news/build.gradle]: Alright, working on this. I assume we are happy with latest Gradle 2.x? Or wanna test out the fresh gradle 3?

Comment (saurfang) [inline demo/twenty_news/build.gradle]: Don't think anyone here has more experience with gradle than you do ;) I don't have a strong opinion either.

**Comment by jq** [comment]
👍
